obal/tito_extensions/git_source_strategy.py

```diff
@@ -0,0 +1,183 @@
+
+        # Replace version in spec:
+        version_regex = re.compile("^(version:\s*)(.+)$", re.IGNORECASE)
+        self.version = self._get_version()
+        print("Building version: %s" % self.version)
+        replacements.append((version_regex, "Version: %s\n" % self.version))
+        self.replace_in_spec(replacements)
+
+        rel_date = datetime.utcnow().strftime("%Y%m%d%H%M")
+        self.release = rel_date + gitrev
+        print("Building release: %s" % self.release)
+        run_command("sed -i '/^Release:/ s/%%/.%s%%/' %s" % (self.release, self.spec_file))
+
+
+    """
+    Generates the source file from a local repo, useful for generating
+    packages of custom branches.
+    It first copies source files from git, then copies the source file
+    over the top, so they're merged (patches etc can then be stored
+    in git).
+    Takes the following arguments:
+      source_dir: path to repo ("~/foreman")
+    """
+    def _fetch_local(self):
+
+        source_dir = os.path.expanduser(self.builder.args['source_dir'][0])
+
+        version_regex = re.compile("^(version:\s*)(.+)$", re.IGNORECASE)
+
+        with open(self.spec_file, 'r') as spec:
+            for line in spec.readlines():
+                match = version_regex.match(line)
+                if match:
+                    version = match.group(2)
+
+        if not version:
+            error_out("Version not found in spec")
+
+        old_dir = os.getcwd()
+        os.chdir(source_dir)
+
+        fetchdir = os.path.join(self.builder.rpmbuild_sourcedir, 'archive')
+        if not os.path.exists(fetchdir):
+          os.mkdir(fetchdir)
+
+        arch_prefix = "-".join([self.builder.project_name, version])
+
+        # git archive --format=tar.gz --prefix=pulp-2.15.0/ master > pulp-2.15.0.tar.gz
+
+        with open("./%s.tar.gz" % arch_prefix, "w+") as archive:
+            subprocess.call(["git", "archive", "--format=tar.gz", ("--prefix=%s/" % arch_prefix), "HEAD"], stdout=archive)
+
+        sources = glob.glob("./*.tar.gz")
+        print(sources)
+
+        for srcfile in sources:
+          debug("Copying %s from local source dir" % srcfile)
+          shutil.move(srcfile, os.path.join(fetchdir, os.path.basename(srcfile)))
+
+        gitrev = "local"
+        gitsha = subprocess.check_output(["git", "rev-parse", "HEAD"]).decode('utf-8')
+        if gitsha:
+          gitrev = "git%s" % gitsha[0:7]
+
+        os.chdir(old_dir)
+
+        return gitrev
+
+    def _get_version(self):
+        """
+        Get the version from the builder.
+        Sources are configured at this point.
+        """
+        # Assuming source0 is a tar.gz we can extract a version from:
+        base_name = os.path.basename(self.sources[0])
+        debug("Extracting version from: %s" % base_name)
+
+        # Example filename: tito-0.4.18.tar.gz:
+        simple_version_re = re.compile(".*-(.*).(tar.gz|tgz|zip|tar.bz2|gem)")
```

Comment:
```suggestion
        simple_version_re = re.compile(".*-(.*).(tar.(gz|bz2|xz)|tgz|zip|gem)")
```

Reply:
This line is pulled again from the aforementioned `custom.py`.  We can change it, but unchanged it has endured for a long period of time there without needing modification.  I'm hesitant to induce such a change without strong need.